server/routes/invitation.js

```diff
@@ -1,26 +1,147 @@
-        const {id} = req.params;
-        const {to_user_id} = req.body;
-        let to_user_email = req.body.to_user_email ? req.body.to_user_email : null;
-        var objId = mongoose.Types.ObjectId;
+        const {toEmailAr, fromEmail, referralId} = req.body;
+        let invalidEmails = [];
+        let validEmails = [];
+        let curUserEmails = [];
+        let nonCurUserEmails = [];
+        let inviteRecipients = [];
+        let inviteCreatedInternalMessage = '';
+        let inviteCreatedEmailMessage = '';
 
-        const invite = new Invitation({
-            "from_user": new objId(id),
-            "to_user": new objId(to_user_id),
-            to_user_email
-        });
-        invite.save(function(err) {
-            if (err) return handleError(err);
-            res.json({ type: "success", message: "The invitation was saved."});
-        });
+        /* email validation */
+        if (toEmailAr.length === 0) {
+            return res.status(400).json({error: 'No email addresses for invitation recipients were provided.'})
+        }
+        toEmailAr.forEach(email => {
+            if (validator.isEmail(email)) {
+                validEmails.push(email);
+            } else {
+                invalidEmails.push(email);
+            }
+        })
+        if (!validEmails.length) {
+            return res.status(400).json({error: 'Email addresses for invitation recipients were invalid. Please check the spelling.'})
+        } else {
+            validEmails.forEach((to_email, idx) => {
+                User.find({email: to_email}, function(err, user) {
+                    if (err) console.error('unable to find user', err);
+                    if (!user.length) {
+                        nonCurUserEmails.push(to_email);
+                    }
+                    if (user.length) {
+                        curUserEmails.push(to_email);
+                    }
+                    if (idx === validEmails.length - 1) {
+                        //1 handle case where recipients include registered users and non registered
+                        if (curUserEmails.length && nonCurUserEmails.length) {
+                            let newInvites = [];
+                            console.log('curUserEmails', curUserEmails)
+                            curUserEmails.forEach(to_user_email => {
+                                let newInvite = {to_user_email, from_user_email: fromEmail};
+                                newInvites.push(newInvite);
+                            });
+                            Invitation.insertMany(newInvites, function(err) {
+                                if (err) return console.error(err);
+                                inviteRecipients = inviteRecipients.concat(curUserEmails);
+                                if (nonCurUserEmails.length === 1) {
+                                    sendEmail({from_email: fromEmail, 
+                                                to_email: nonCurUserEmails[0], 
```

Comment:
since theres the `to_email_ar`, and `nonCurUserEmails` is already an array, is it possible to just send `nonCurUserEmails` into `sendEmailMultiple` without worrying about if its length 1 or more?